### PR TITLE
BOAC-1293: Returns 500 status when cache refresh fails

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-1293

### CODE REVIEW

File: tests/test_api/test_admin_controller.py [hunk 1]
```diff
@@ -77,6 +76,13 @@ def test_api_key_disabled(self, app, client):
         response = client.get('/api/admin/cachejob', headers=headers)
         assert response.status_code == 401
 
+class TestCachejobRefresh:
+    """Cachejob Refresh API."""
+    def test_cachejob_refresh_success(self, client, admin_session):
+        """Returns a well-formed sucess response."""
+        response = client.get('/api/admin/cachejob/refresh')
+        assert response.status_code == 200
+        assert response.headers.get('Content-Type') == 'application/json'
```
Comment (lyttam) [inline tests/test_api/test_admin_controller.py]: I wanted to write a test for the error response, but I don't think it's possible with our current mocking capabilities.

File: boac/api/admin_controller.py [hunk 2]
```diff
@@ -85,7 +85,11 @@ def start_load_only():
 @app.route('/api/admin/cachejob/refresh')
 @admin_required
 def start_refresh():
-    return tolerant_jsonify(cache_utils.refresh_request_handler(term()))
+    response = cache_utils.refresh_request_handler(term())
+    if response.get('error'):
+        return tolerant_jsonify(response, status=500)
+
+    return tolerant_jsonify(response)
```
Comment (johncrossman) [inline boac/api/admin_controller.py]: Looks good @lyttam. Sort out Travis woes and I'll merge. Fyi, the above could be written with a one-liner:
```
return tolerant_jsonify(response, status=500 if 'error' in response else 200)
```

**Comment by paulkerschen** [comment]
@lyttam Travis's lint-py complaints should be trivial to fix. Unfortunately, I suspect the one pytest failure is caused by the new cache job refresh test. Since we aren't able to mock out that endpoint, hitting it really does trigger a cache job refresh in a background thread, which in turn creates a bunch of alerts that conflict with later code trying to set up alerts of its own.

Nessie deals with a similar problem by suppressing background job kickoff in test mode.
https://github.com/ets-berkeley-edu/nessie/blob/master/nessie/jobs/background_job.py#L88
Alternatively, we could just wait for a functional mocking library before testing that endpoint.

File: boac/api/admin_controller.py (hunk 2 above)
Comment (lyttam) [reply boac/api/admin_controller.py]: Nice! Will amend.

**Comment by lyttam** [comment]
@johncrossman Amended.

**Comment by johncrossman** [comment]
Thanks @lyttam

**Pull request merged**